src/llm/io_processing/qwen3coder/qwen3coder_tool_parser.cpp

```diff
@@ -127,77 +114,115 @@ static const ParametersTypeMap_t parseToolSchema(const std::string& functionName
-            // escaping '/' is for now omitted as its creating issues in Continue
-
-        /*case '\\':
-            output += "\\\\";
-            break;*/
-        case '\b':
-            output += "\\b";
-            break;
-        case '\f':
-            output += "\\f";
-            break;
-        case '\t':
-            output += "\\t";
-            break;
-        default:
-            output += c;
-        }
-    }
-    return output;
-}
-
-static std::string setCorrectValueType(std::string& inputValue, const std::string& currentParameterName, const ParametersTypeMap_t& parametersType) {
-    auto paramIt = parametersType.find(currentParameterName);
-    if (paramIt == parametersType.end()) {
-        SPDLOG_DEBUG("Parameter: {} schema not found , leaving as is", currentParameterName);
-        return inputValue;
-    }
-    if (paramIt->second == ParameterType::STRING) {
-        inputValue = "\"" + escapeQuotes(inputValue) + "\"";
-        return inputValue;
-    }
-    if (paramIt->second == ParameterType::BOOLEAN) {
-        // in case of bool we need to convert to lower case
-        std::transform(inputValue.begin(), inputValue.end(), inputValue.begin(), ::tolower);
-        return inputValue;
-    }
-    return inputValue;
-}
-
 #define DEFINE_TAG_POSITION_AND_BREAK_IF_NOT_FOUND(TAG)                         \
     auto pos = this->streamContent.find(TAG, this->getLastProcessedPosition()); \
     if (pos == std::string::npos) {                                             \
         SPDLOG_TRACE("Did not find: {}", TAG);                                  \
         break;                                                                  \
     }
 
+static const char* jsonTypeOf(const rapidjson::Value& val) {
+    if (val.IsObject())
+        return "object";
+    if (val.IsArray())
+        return "array";
+    if (val.IsString())
+        return "string";
+    if (val.IsBool())
+        return "bool";
+    if (val.IsInt())
+        return "int";
+    if (val.IsUint())
+        return "uint";
+    if (val.IsInt64())
+        return "int64";
+    if (val.IsUint64())
+        return "uint64";
+    if (val.IsDouble())
+        return "double";
+    if (val.IsNumber())
+        return "number";
+    if (val.IsNull())
+        return "null";
+    return "unknown";
+}
+static void enforceStringValue(rapidjson::Value& v,
+    rapidjson::Document::AllocatorType& alloc) {
+    if (v.IsString()) {
+        return;
+    }
+    std::string s;
+    if (v.IsBool()) {
+        s = v.GetBool() ? "true" : "false";
+    } else if (v.IsInt()) {
+        s = std::to_string(v.GetInt());
+    } else if (v.IsUint()) {
+        s = std::to_string(v.GetUint());
+    } else if (v.IsInt64()) {
+        s = std::to_string(v.GetInt64());
+    } else if (v.IsUint64()) {
+        s = std::to_string(v.GetUint64());
+    } else if (v.IsDouble()) {
+        s = std::to_string(v.GetDouble());
+    } else if (v.IsNull()) {
+        s = "null";
+    } else {
```

Comment:
Is this else block for arrays and objects? If it works for them, can't we use this part also for simple types Int, Uint, Int64 etc. where you currently use `to_string` call?